### PR TITLE
[Modules Restructuring] Restructure Editor Module

### LINKED ISSUE #16564: Modules Restructuring - Editor Module
Parent #16562
Related Repo [WordPress-Editor-Android](https://github.com/wordpress-mobile/WordPress-Editor-Android)

This issue is about investigating the `editor` lib project module and converting it to pure module.

- [x] Investigate module @ParaskP7
  - [x] `⛔️ [DEPRECATED]` prefix added to the `About` section of the [related repo](https://github.com/wordpress-mobile/WordPress-Editor-Android).
  - [x] Archive [related repo](https://github.com/wordpress-mobile/WordPress-Editor-Android) and make it read-only ❓ 🤔   (just like it is done for the [networking](https://github.com/wordpress-mobile/WordPress-Networking-Android) repo)
- [x] Convert to pure module @ParaskP7 #16769

### PR DESCRIPTION
**Pull request by ParaskP7**
Parent #16562
Fixes #16564

This PR restructures the `editor` module by disassociating the [WordPress-Editor-Android](https://github.com/wordpress-mobile/WordPress-Editor-Android) repo and removing the extra `WordPressEditor` level from it.  This effectively makes `editor` a pure module, without any extra inner levels or unnecessary build files lying around (like `settings.gradle`, `.gitignore`, etc).

Also, this PR activates the `editor-example` module and placing that on the same level as the `editor` module.

PS.1: @antonis I added you as the main reviewer, that is, in addition to the @wordpress-mobile/owl-team team itself, since you helped us with the [OPA Android](https://github.com/Automattic/opa-android) project a while back, which involved working the `editor` and testing it, starting with the `editor-example`. As such, I think you might be the best person to sign-off on that change for WPAndroid. 🥇

PS.2 @antonis as part of the above, please take a moment to also sign-off the `editor-example` module changes as well (especially [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/054b97523060cb0613714eae32d0867943955dd3) and [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/bad8f188fa0bf5e85be459c30d18916437129935)). With this change you will notice that this `editor-example` module is now active and buildable too, still, it fails to compile as there are `21` errors with the code itself. My intention with this PR is to ignore those compilation errors for now and maybe come back to fixing them later on with a separate PR, so don't worry too much about them, I'll add a task on it within the [parent](https://github.com/wordpress-mobile/WordPress-Android/issues/16562) ticket afterwards. You can try assembling this module by running the `./gradlew libs:editor-example:assembleDebug` command, but FYI, this doesn't affect building the main WPAndroid app anyway. Thus, it failing doesn't matter much, that is in terms of progressing this PR. Also, note that our CI or any of our processes are unaware of it, this is just a nice to have at this point.

PS.3: @AliSoftware I added you as a secondary reviewer, that is, in addition to the @wordpress-mobile/owl-team itself, as I did another [localization.rb](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/61a20737c220914d61cf8bbab03134c519f81030) related change that I want you to double check and verify again, just in case I missed something vital. Also, in addition to that, please double check this minor [strings.xml](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/edd52f2330bf9e13f108ea2d8bf978fdfac3598d) related change that update the string group section comment.

-----

Question (❓): As per my licensing question on this other related PR of mine [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16744) and since I saw that the `LICENCE` file for this module is `GPLv2`, unlike the `analytics` mode, which is pointing to `MIT`, I deleted the licence for this module to start depending solely on the root level `LICENCE` that WPAndroid is using, which is `GPLv2` as well, wdyt? 🤔

-----

To test:
- Build the app and install it on your device or emulator.
- Go to `Posts` -> Click on the plus 🔵 FAB button on the bottom right -> Select `Blog post` to create a new post.
- Randomly pick and add few blocks to the post and verify that the post editing/publishing works as expected.

The above is just one example of how this change can tested so that we could quickly verify the overall correctness of this PR. However, feel free to smoke test other aspects as well, especially if you are aware any or all the other places where the `editor` is being used like `Pages` or `Stories`.

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by antonis** [comment]
Thank you for wrangling this @ParaskP7 🙇 
I hope that I will be able to review this later today or tomorrow.
Also pinging @hypest and @fluiddot from Sparrow since this involves the `gutenberg-mobile` integration.

**Comment by ParaskP7** [comment]
Thank you so much @antonis ! 🌟

File: WordPress/src/main/res/values/strings.xml
Comment (AliSoftware) [inline WordPress/src/main/res/values/strings.xml:4108]: 👍

File: config/lint.xml
Comment (AliSoftware) [inline config/lint.xml:20]: Is this something we plan to fix soon? (but in a separate PR though)

File: fastlane/lanes/localization.rb
Comment (AliSoftware) [inline fastlane/lanes/localization.rb:285]: 👍  Matches the new path to this `strings.xml` file 👍

File: fastlane/resources/values/strings.xml
Comment (AliSoftware) [inline fastlane/resources/values/strings.xml:4100]: 👍  (although this would have been auto-updated during next code freeze when our automation copies the `WordPress/src/main/res/values/strings.xml` file to this `fastlane/resources/values/strings.xml` to "freeze" it… but doesn't hurt to manually update it here anyway)

**Review by AliSoftware — COMMENTED** [review]
From a tooling perspective, especially `fastlane/lanes/localization.rb` and the changes in the `strings.xml` files, this LGTM.

Not submitting my review as a formal "Approval" though because I only focused on checking the diff in GitHub for those tooling-related files, but didn't review that the restructuring of the rest of the Android files make sense (since I'm not comfortable enough with Android's typical project structures and directory org conventions to validate those myself) nor did I go thru the testing steps you suggest in the PR description.

File: libs/editor-example/src/main/res/values/colors.xml
Comment (antonis) [inline libs/editor-example/src/main/res/values/colors.xml:1]: I wonder why the 1st line is removed here?

**Review by antonis — APPROVED** [review]
Great work @ParaskP7 👍 
I've tested a AS debug build and the CI Jetpack build on a Pixel 5 (Android 12) and everything worked as expected for me. I also tested using *Metro* at the head of `gutenberg-mobile` `trunk` and didn't notice any changed behavior.

> please take a moment to also sign-off the editor-example module changes as well (especially [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/054b97523060cb0613714eae32d0867943955dd3) and [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/bad8f188fa0bf5e85be459c30d18916437129935))

The code and structure changes also look consistent to me 🎉   

> My intention with this PR is to ignore those compilation errors for now and maybe come back to fixing them later on with a separate PR.

That sounds like a good plan to me 👍 

ps. Since I'm not working with the editor for a few months now I think it would be good to have some eyes from the Sparrow team before merging the changes (cc @fluiddot, @hypest)

**Comment by fluiddot** [comment]
> ps. Since I'm not working with the editor for a few months now I think it would be good to have some eyes from the Sparrow team before merging the changes (cc @fluiddot, @hypest)

@ParaskP7 I've run some tests by running the app locally and everything seems to work fine. I also tested the installable builds and haven't found any issue 🎊. Being this said, from the editor's perspective, the changes look great.

File: config/lint.xml
Comment (ParaskP7) [reply config/lint.xml:20]: This is one of the many warnings that we are having, as such I wouldn't bet on it being planned to be fixed soon. You will notice that I just moved this warning from one place (by removing the `editor` related `lint.xml` file) to this common place (see [commit](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/bb4a2d775abd1e2d5387ae9027dd676ecdac8580) description).

As such, I wouldn't worry about fixing this warning, at least as part of this group of PRs or in general, this project. This kind of warnings should be fixed as part of a more generic Lint project. Hopefully, we will get to that later on and sooner than later. 🤞

Does that makes sense @AliSoftware ?

**Comment by ParaskP7** [comment]
👋 @AliSoftware !

> From a tooling perspective, especially fastlane/lanes/localization.rb and the changes in the strings.xml files, this LGTM.

Thank you for your input, your 👍 and your review comments! 🙇

File: libs/editor-example/src/main/res/values/colors.xml
Comment (ParaskP7) [reply libs/editor-example/src/main/res/values/colors.xml:1]: 👋 @antonis !

Yea, it was failing CheckStyle (see [commit](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/45bc857912283342d839043bc109121711641ad3)) and as such I removed this line. This line is actually and in general not required anyway.

With this change I also made this `res/values` XML files consistent with each other, as none of them had this line as well. As such, this was another good reason for me to remove this line, process with this change and never look back... 😅

**Comment by ParaskP7** [comment]
👋 @antonis !

Thank you for reviewing, testing this PR and the 👍 ! 🙇 

Also, thank you for giving me the 👍 on the `editor-example` change and for tagging Carlos. 🙏

**Comment by ParaskP7** [comment]
👋 @fluiddot !

Thank you so much for reviewing and testing this PR as well, you rock! 🙇

**Comment by ParaskP7** [comment]
👋 @oguzkocer !

I am not sure if you also want to double-check on this PR, at least on the Gradle changes like you did with the rest, and give a 👍 as well before I merge this to `trunk`. In case you do, I am going to keep this PR open for at least today. Thank YOU so much! 🙏 

PS: Apologies for the shameless ping... 😊

**Review by oguzkocer — APPROVED** [review]
@ParaskP7 `./gradlew :libs:editor-example:assemble` is currently failing. I don't think this is because of the changes made in this PR, so I am still approving the PR to keep things going.

[It looks like the last non build related change to `libs/editor/example` was 3 years ago](https://github.com/wordpress-mobile/WordPress-Android/commits/trunk/libs/editor/example). If this module is not used and doesn't have a lot of value, I think we can consider removing it. On the other hand, if there is value, we should at least add it's `assemble` task to Buildkite, so it doesn't get outdated to the point that it doesn't build anymore.

**Comment by ParaskP7** [comment]
👋 @oguzkocer !

> @ParaskP7 ./gradlew :libs:editor-example:assemble is currently failing. I don't think this is because of the changes made in this PR, so I am still approving the PR to keep things going.

Spot on, yes, thank you! 👍 

> [It looks like the last non build related change to libs/editor/example was 3 years ago](https://github.com/wordpress-mobile/WordPress-Android/commits/trunk/libs/editor/example).

👍

> If this module is not used and doesn't have a lot of value, I think we can consider removing it.

Yes, I was thinking about that too. @fluiddot @antonis what are your thoughts on that? Does this module provide any value to you, are you utilizing it somehow? 🤔 

Personally, I only remember that the last time this module provided any real value to us was with the [OPA Android](https://github.com/Automattic/opa-android) project we did a while back where @antonis tried to make sense of the this `editor-example` module and setup something similar for OPA's editor configuration as well (by checking on the `EditorExampleActivity`). This was because, as I remember, doing the same using the WordPress's editor configuration (by checking on the `EditorPostActivity`) was too complicated. Isn't that right? 🤔 

> On the other hand, if there is value, we should at least add it's assemble task to Buildkite, so it doesn't get outdated to the point that it doesn't build anymore.

Yes, definitely, in case we decide to keep this `editor-example` module, we should add its assemble task to Buildkite, I totally agree! 💯 

-----

FYI: I'll add some tasks on the [parent issue](https://github.com/wordpress-mobile/WordPress-Android/issues/16562) so that we don't forget to proceed with the `editor-example` related work, no matter if we decide to keep and update it, or remove it altogether.

**Comment by antonis** [comment]
> Does this module provide any value to you, are you utilizing it somehow?

@ParaskP7 My view is that this is indeed not used at this point and has not been maintained for long. The only usage I found as you mentioned was as an example for the OPA project. On my side I feel that it is safe to remove it.

**Comment by ParaskP7** [comment]
Thank you for your input @antonis ! 🥇 

> @ParaskP7 My view is that this is indeed not used at this point and has not been maintained for long.

Great, then we have:
- 3/4 votes in favor of removing this `editor-example` module. @oguzkocer @antonis @ParaskP7 
- 1/4 votes waiting for confirmation in favor or against removing this `editor-example` module. @fluiddot 

> The only usage I found as you mentioned was as an example for the OPA project. On my side I feel that it is safe to remove it.

Coolio, and we can always check-out on the archived [WordPress-Editor-Android](https://github.com/wordpress-mobile/WordPress-Editor-Android) repo if we ever need a simple example of an editor related configuration. This [example](https://github.com/wordpress-mobile/WordPress-Editor-Android/tree/develop/example) module and its [EditorExampleActivity](https://github.com/wordpress-mobile/WordPress-Editor-Android/blob/develop/example/src/main/java/org/wordpress/android/editor/example/EditorExampleActivity.java) editor configuration will always be there if we even need it again. 🥇

**Comment by fluiddot** [comment]
> Great, then we have:
> 
> * 3/4 votes in favor of removing this `editor-example` module. @oguzkocer @antonis @ParaskP7
> * 1/4 votes waiting for confirmation in favor or against removing this `editor-example` module. @fluiddot

Sorry for the delay, I saw that the unit tests [are not being invoked in the CI jobs](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/.buildkite/pipeline.yml), and the files weren't updated since a while ago. So, I'd also vote in favor of removing it ✂️ .

**Comment by ParaskP7** [comment]
Thank you for your input @fluiddot and no delay at all, actually you were super fast instead! 🙇 

> I saw that the unit tests [are not being invoked in the CI jobs](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/.buildkite/pipeline.yml), and the files weren't updated since a while ago. So, I'd also vote in favor of removing it ✂️ .

Great, it is then sorted, I will be deleting this `editor-example` module in a separate PR after merging this into `trunk`. 🥇

**Pull request merged**
